marketplace/image-focal-point/src/components/AppView/AppView.js

```diff
@@ -10,129 +10,109 @@ import { styles } from './styles';
 
 import appLogo from './app-logo.svg';
 
-const APP_INSTALLATION_STATE = {
-  LOADING: 'LOADING',
-  INSTALLED: 'INSTALLED',
-  NOT_INSTALLED: 'NOT_INSTALLED'
-};
-
-function convertToAppInstallationState(appIsInstalled) {
-  return appIsInstalled ? APP_INSTALLATION_STATE.INSTALLED : APP_INSTALLATION_STATE.NOT_INSTALLED;
-}
+const makeContentType = (contentTypeId, contentTypeName) => ({
+  sys: {
+    id: contentTypeId
+  },
+  name: contentTypeName,
+  displayField: 'title',
+  fields: [
+    {
+      id: 'title',
+      name: 'Title',
+      required: true,
+      type: 'Symbol'
+    },
+    {
+      id: 'image',
+      name: 'Image',
+      required: true,
+      type: 'Link',
+      linkType: 'Asset'
+    },
+    {
+      id: 'focalPoint',
+      name: 'Focal point',
+      required: true,
+      type: 'Object'
+    }
+  ]
+});
 
 export class AppView extends Component {
   static propTypes = {
     sdk: PropTypes.object.isRequired
   };
 
   state = {
+    isInstalled: false,
     allContentTypesIds: [],
-    appInstallationState: APP_INSTALLATION_STATE.LOADING,
     contentTypeId: camelCase('Image with Focal Point'),
     contentTypeName: 'Image with Focal Point',
     isContentTypeIdPristine: true
   };
 
   async componentDidMount() {
-    const { sdk } = this.props;
-    const { app } = sdk.platformAlpha;
+    const { space, app } = this.props.sdk;
 
-    const appIsInstalled = await app.isInstalled();
+    const [isInstalled, allContentTypes] = await Promise.all([
+      app.isInstalled(),
+      space.getContentTypes()
+    ]);
 
-    const allContentTypes = await sdk.space.getContentTypes();
     const allContentTypesIds = allContentTypes.items.map(({ sys: { id } }) => id);
 
     // Following eslint error is caused due to using async/await
     // eslint-disable-next-line react/no-did-mount-set-state
-    this.setState(
-      {
-        appInstallationState: convertToAppInstallationState(appIsInstalled),
-        allContentTypesIds
-      },
-      () => {
-        sdk.platformAlpha.app.setReady();
-      }
-    );
+    this.setState({ isInstalled, allContentTypesIds }, () => app.setReady());
 
-    if (!appIsInstalled) {
-      app.onConfigure(this.installApp);
-    }
+    app.onConfigure(this.installApp);
+    app.onConfigurationCompleted(err => {
+      if (!err) {
+        this.setState({ isInstalled: true });
+      }
+    });
   }
 
   installApp = async () => {
-    const { allContentTypesIds, contentTypeId, contentTypeName } = this.state;
+    const { sdk } = this.props;
+    const { isInstalled, allContentTypesIds, contentTypeId, contentTypeName } = this.state;
+
+    if (isInstalled) {
+      sdk.notifier.error('Nothing to do: the app is already fully configured.');
```

Comment:
What about just removing this message since the user doesn't need to know that nothing happened. 

Or change it from error to the green message block since there is no error.

Reply:
True, will make it green